contributors/devel/sig-testing/flaky-tests.md

```diff
@@ -140,34 +158,242 @@ release-blocking flakes. Therefore we have the following guidelines:
+- Be sure to tag the relevant SIG, if you know what it is.
+
+For a good example of a flaking test issue,
+[check here](https://github.com/kubernetes/kubernetes/issues/93358).
+
+## Reproducing flaky test results
+
+For more information about deflaking Kubernetes tests, watch @liggitt's
+[presentation from Kubernetes SIG Testing - 2020-08-25](https://www.youtube.com/watch?v=Ewp8LNY_qTg).
+
+### Reproducing unit test flakes
+
+Start with a simple attempt to just run the flaky unit test. For example:
+
+```sh
+go test ./pkg/kubelet/config -run TestInvalidPodFiltered
 ```
-$ go install golang.org/x/tools/cmd/stress
+
+Also make sure that you bypass the `go test` cache by using an uncachable
+command line option:
+
+```sh
+go test ./pkg/kubelet/config -count=1 -run TestInvalidPodFiltered
+```
+
+If even this is not revealing issues with the flaky test, try running with
+[race detection](https://golang.org/doc/articles/race_detector.html) enabled:
+
+```sh
+go test ./pkg/kubelet/config -race -count=1 -run TestInvalidPodFiltered
+```
+
+Finally, you can stress test the unit test using the
+[stress command](https://godoc.org/golang.org/x/tools/cmd/stress). Install it
+with this command: 
+
+```sh
+go get golang.org/x/tools/cmd/stress
 ```
 
-Then build your test binary
+Then build your test binary:
 
+```sh
+go test ./pkg/kubelet/config -race -c
 ```
-$ go test -c -race
+
+Then run it under stress:
+
+```sh
+stress ./config.test -test.run TestInvalidPodFiltered
 ```
 
-Then run it under stress
+The stress command runs the test binary repeatedly, reporting when it fails. It
+will periodically report how many times it has run and how many failures have
+occurred. 
+
+You should see output like this:
 
 ```
-$ stress ./package.test -test.run=FlakyTest
+411 runs so far, 0 failures
+/var/folders/7f/9xt_73f12xlby0w362rgk0s400kjgb/T/go-stress-20200825T115041-341977266
+--- FAIL: TestInvalidPodFiltered (0.00s)
+    config_test.go:126: Expected no update in channel, Got types.PodUpdate{Pods:[]*v1.Pod{(*v1.Pod)(0xc00059e400)}, Op:1, Source:"test"}
+FAIL
+ERROR: exit status 1
+815 runs so far, 1 failures
 ```
 
-It runs the command and writes output to `/tmp/gostress-*` files when it fails.
-It periodically reports with run counts. Be careful with tests that use the
-`net/http/httptest` package; they could exhaust the available ports on your
-system!
+Be careful with tests that use the `net/http/httptest` package; they could
+exhaust the available ports on your system!
+
+### Reproducing flakes using integration tests
+
+Integration tests run similarly to unit tests, but they almost always expect a
+running `etcd` instance. You should already have `etcd` installed if you have
+followed the instructions in the [Development Guide](../development.md). Run
+`etcd` in another shell window or tab.
+
+Compile your integration test using a command like this:
+
+```sh
+https://github.com/kubernetes/kubernetes/issues/93496#issuecomment-678375312
```

Comment:
that link isn't pointing at a compile command...

Reply:
Oh, good catch. I must have had the wrong thing in my paste buffer. I will fix!